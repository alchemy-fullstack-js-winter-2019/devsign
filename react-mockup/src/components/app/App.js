import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from '../Home';
import HumFeed from '../hums/HumFeed';
import UserDetail from '../user/UserDetail';
import Callback from '../../containers/auth/Callback';
import { withSession } from '../../containers/auth/withSession';
import Header from './Header';

function App() {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={withSession(Home)} />
        <Route exact path="/hum-feed" component={withSession(HumFeed)} />
        <Route exact path="/profile" component={withSession(UserDetail)} />
        <Route exact path="/header" component={Header} />
        <Route exact path="/callback" component={Callback} />
      </Switch>
    </Router>
  );
}

export default App;
