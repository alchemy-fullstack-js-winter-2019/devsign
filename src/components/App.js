import React, { PureComponent } from 'react';
import Header from './Header';
import Footer from './Footer';



export default class App extends PureComponent {
 
  render() {
    return (
      <>
      <Header />
      <Footer />
      </>
    );
  }

}
