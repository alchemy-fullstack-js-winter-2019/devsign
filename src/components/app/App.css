body {
  background-color: rgb(16, 137, 158);
  text-align: center;
}

img {
  width: 55px;
}