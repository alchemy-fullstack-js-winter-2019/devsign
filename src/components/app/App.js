import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import Header from '../Header';
import Home from '../../containers/home/Home';
import SignPage from '../../containers/register/SignPage';
import TopTweets from '../../containers/tweets/TopTweets';
import UserDisplay from '../../containers/users/UserDispaly';
import './App.css';

export default function App() {
  return (
    <Router>
      <>
        <Header />
        <Switch>
          <Route path="/user/:id" component={UserDisplay}></Route>
          <Route path="/toptweets" component={TopTweets}></Route>
          <Route path="/register" component={SignPage}></Route>
          <Route path="/" component={Home}></Route>
        </Switch>
      </>
    </Router>
  );
}
