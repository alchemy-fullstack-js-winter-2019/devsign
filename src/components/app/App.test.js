import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

jest.mock('../../services/auth.js', () => ({}));

describe('App component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <App />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
