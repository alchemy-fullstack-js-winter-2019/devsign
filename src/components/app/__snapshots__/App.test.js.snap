// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches a snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <nav>
      <nav
        className="nav"
      >
        <ul>
          <a
            href="/home"
            onClick={[Function]}
          >
            <li>
              Home
            </li>
          </a>
          <a
            href="/profile"
            onClick={[Function]}
          >
            <li>
              Profile
            </li>
          </a>
          <a
            href="/twitter"
            onClick={[Function]}
          >
            <li>
              My Twitter Page
            </li>
          </a>
        </ul>
      </nav>
    </nav>
  </header>,
  <main
    className="main"
  >
    <h1>
      Welcome to your Tweety Tweets
    </h1>
    <h2>
      Just another way to staying relevant!
    </h2>
    <div
      className="register"
    >
      <a
        href="#0"
      >
        Sign up
      </a>
      <a
        href="#0"
      >
        Log in
      </a>
    </div>
  </main>,
]
`;
